src/System.Management.Automation/help/HelpProvider.cs

```diff
@@ -226,7 +226,7 @@ internal string GetDefaultShellSearchPath()
             string shellID = this.HelpSystem.ExecutionContext.ShellID;
             // Beginning in PowerShell 6.0.0.12, the $pshome is no longer registry specified, we search the application base instead.
             // We use executing assemblies location in case registry entry not found
-            return Utils.GetApplicationBase(shellID) ?? Path.GetDirectoryName(Environment.ProcessPath);
+            return Utils.GetApplicationBase() ?? Path.GetDirectoryName(Environment.ProcessPath);
```

Comment:
This fallback is used in AppContext.BaseDirectory for NATIVEAOT(wasm).
I guess it is right fallback in common and we should be use it in GetApplicationBase() and ask .Net team to implement this in .Net 8.

Reply:
We're already using this fallback, so let's use this for now.  I'll update the code and cleanup this piece as well.